refactor(examples) Refactor quickstart-sklearn-tabular to use the new Flower (#3893)
adap/flower: examples/quickstart-sklearn-tabular/README.md

Reviewer: We need to add a `title:` to the metadata at the top and update the title of the readme to the same title in the `pyprojec.toml`: `Federated learning with scikit-learn and Flower`

Author: Solved

---

> title: Federated Learning with scikit-learn and Flower

Reviewer: ```suggestion
```

Reviewer: in the end we decided to remove this.

tags: [quickstart, tabular, fds]
dataset: [Iris]
framework: [scikit-learn]
---

> # Federated Learning with scikit-learn and Flower

Reviewer: ```suggestion
# Federated Learning with scikit-learn and Flower (Quickstart Example)
```

Running this example in itself is quite easy. This example uses [Flower Datasets](https://flower.ai/docs/datasets/) to
download, partition and preprocess the dataset.

## Set up the project

### Clone the project

Start by cloning the example project. We prepared a single-line command that you can copy into your shell which will checkout the example for you:

```shell
git clone --depth=1 https://github.com/adap/flower.git && mv flower/examples/quickstart-sklearn-tabular . && rm -rf flower && cd quickstart-sklearn-tabular
```

This will create a new directory called `quickstart-sklearn-tabular` with the following structure:

```shell
quickstart-sklearn-tabular
├── sklearnexample
│   ├── __init__.py
│   ├── client_app.py   # Defines your ClientApp
│   ├── server_app.py   # Defines your ServerApp
│   └── task.py         # Defines your model, training and data loading
├── pyproject.toml      # Project metadata like dependencies and configs
└── README.md
```

### Install dependencies and project

Install the dependencies defined in `pyproject.toml` as well as the `sklearnexample` package.

```bash
pip install -e .
```

## Run the project

> You can run your `ClientApp` and `ServerApp` in both _simulation_ and _deployment_ mode without making changes to the code. If you are starting with Flower, we recommend you using the _simulation_ model as it requires fewer components to be launched manually. By default, `flwr run` will make use of the Simulation Engine.

Reviewer: ```suggestion
You can run your Flower project in both _simulation_ and _deployment_ mode without making changes to the code. If you are starting with Flower, we recommend you using the _simulation_ mode as it requires fewer components to be launched manually. By default, `flwr run` will make use of the Simulation Engine.
```

```

This will create a new directory called `quickstart-sklearn-tabular` with the following structure:

```shell
quickstart-sklearn-tabular
├── sklearnexample
│   ├── __init__.py
│   ├── client_app.py   # Defines your ClientApp
│   ├── server_app.py   # Defines your ServerApp
│   └── task.py         # Defines your model, training and data loading
├── pyproject.toml      # Project metadata like dependencies and configs
└── README.md
```

### Install dependencies and project

Install the dependencies defined in `pyproject.toml` as well as the `sklearnexample` package.

```bash
pip install -e .
```

## Run the project

You can run your Flower project in both _simulation_ and _deployment_ mode without making changes to the code. If you are starting with Flower, we recommend you using the _simulation_ mode as it requires fewer components to be launched manually. By default, `flwr run` will make use of the Simulation Engine.

### Run with the Simulation Engine

```bash
flwr run .
```

You can also override some of the settings for your `ClientApp` and `ServerApp` defined in `pyproject.toml`. For example:

```bash

> flwr run . --run-config penalty="l1"

Reviewer: ```suggestion
flwr run . --run-config num-server-rounds =10
```

Reviewer: it seems "l1" penalty is not supported, so i'd suggest showing the overriding functionality with another setting.

```
ValueError: Solver lbfgs supports only 'l2' or None penalties, got l1 penalty.
```

Author: I've changed the solver to "saga" now which supports also `l1`.

Though I cannot run the 
`flwr run . --run-config penalty="l1"`

`CalledProcessError: Command '['flower-simulation', '--app', '.', '--num-supernodes', '3', '--run-config', 'penalty=l1']' returned non-zero exit status 1.`

Running this example in itself is quite easy. This example uses [Flower Datasets](https://flower.ai/docs/datasets/) to
download, partition and preprocess the dataset.

## Set up the project

### Clone the project

Start by cloning the example project. We prepared a single-line command that you can copy into your shell which will checkout the example for you:

```shell
git clone --depth=1 https://github.com/adap/flower.git && mv flower/examples/quickstart-sklearn-tabular . && rm -rf flower && cd quickstart-sklearn-tabular
```

This will create a new directory called `quickstart-sklearn-tabular` with the following structure:

```shell
quickstart-sklearn-tabular

> ├── example

Reviewer: ```suggestion
├── sklearnexample
```

│   ├── __init__.py
│   ├── client_app.py   # Defines your ClientApp
│   ├── server_app.py   # Defines your ServerApp
│   └── task.py         # Defines your model, training and data loading
├── pyproject.toml      # Project metadata like dependencies and configs
└── README.md
```

### Install dependencies and project

> Install the dependencies defined in `pyproject.toml` as well as the `mlxexample` package.

Reviewer: ```suggestion
Install the dependencies defined in `pyproject.toml` as well as the `sklearnexample` package.
```

```bash
pip install -e .
```

## Run the project

You can run your `ClientApp` and `ServerApp` in both _simulation_ and _deployment_ mode without making changes to the code. If you are starting with Flower, we recommend you using the _simulation_ model as it requires fewer components to be launched manually. By default, `flwr run` will make use of the Simulation Engine.

### Run with the Simulation Engine

```bash
flwr run .
```

You can also override some of the settings for your `ClientApp` and `ServerApp` defined in `pyproject.toml`. For example:

```bash
flwr run . --run-config penalty="l1"
```

### Run with the Deployment Engine

> \[!NOTE\]
> An update to this example will show how to run this Flower application with the Deployment Engine and TLS certificates, or with Docker.

## Explanation

This example is a federated version of the centralized case that can be found

> [here](https://github.com/ml-explore/mlx-examples/tree/main/mnist).

Reviewer: remove all this also?

Author: Solved
